src/components/ProfileCard/BasicDetailsEdit/hooks.ts

```diff
@@ -64,7 +71,46 @@ const useUpdateProfileDetails = () => {
     }
   };
 
+
+
   return updateProfileWorkplace;
 };
 
-export default useUpdateProfileDetails;
+interface IReturnType {
+  data: any;
+  isLoading: boolean;
+  refetch: () => any;
+}
+
+const fetchNFTs = (walletId: any, agree: boolean): IReturnType => {
+  const chain = 'polygon';
+  const { data, isLoading, refetch } = useQuery(
+    'fetchNFTs',
+    async () => {
+      const response = await fetch(
+        `https://deep-index.moralis.io/api/v2/${walletId}/nft?chain=${chain}`,
+        {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+            'X-API-Key': import.meta.env.VITE_MORALIS_API_KEY || '',
+          },
+        }
+      );
+      const res = await response.json();
```

Comment:
I think here we're missing the part if response is `ok` or not. We can add that function which checks it I guess